### PR TITLE
Clear error after successful request

### PR DESCRIPTION
**Pull request by p3drosola**
The collection.error object is hanging around even after a successful requests. Does it make sense to clear it or did I misunderstand something?

### CODE REVIEW

File: __tests__/Collection.spec.js [hunk 1]
```diff
@@ -163,6 +173,15 @@ describe('Collection', () => {
             expect(collection.error.body).toBe(error)
           }
         })
+        it('clears the error', async () => {
+          reject()
```
Comment (masylum) [inline __tests__/Collection.spec.js]: no need for that one, no? it's already on the `beforeEach`

File: __tests__/Collection.spec.js [hunk 2]
```diff
@@ -123,6 +123,16 @@ describe('Collection', () => {
           }
         })
 
+        it('clears the error', async () => {
+          reject()
```
Comment (masylum) [inline __tests__/Collection.spec.js]: ditto

File: __tests__/Collection.spec.js [hunk 3]
```diff
@@ -211,6 +230,15 @@ describe('Collection', () => {
...
         expect(collection.at(1).get('name')).toBe('bob')
       })
+
+      it('clears the error', async () => {
+        try {
+          await collection.fetch()
+        } catch (e) {}
+        resolve([item, { id: 2, name: 'bob' }])()
+        await collection.fetch()
+        expect(collection.error).toBe(null)
```
Comment (masylum) [inline __tests__/Collection.spec.js]: not sure about this test, it never rejects

**Comment by p3drosola** [comment]
You're right, I didn't really take time to understand how they were set up.

What do you think about this approach?

**Comment by p3drosola** [comment]
Just checking in here. Do you want something changed in the specs? Are you unconvinced the whole approach?

**Comment by masylum** [comment]
Looks good! Sorry if I answer late, I was holidaying 🌴 😎

**Pull request merged**